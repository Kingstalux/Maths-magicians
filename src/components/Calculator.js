import React, { useState } from 'react';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
    display: '0',
  });
  const handleClick = (e) => {
    const {
      total, next, operation, display,
    } = state;
    const result = calculate({ total, next, operation }, e.target.value);
    setState(() => ({
      total: result.total,
      next: result.next,
      operation: result.operation,
      display: display === '0' ? `${`${e.target.value}`}` : `${display} ${e.target.value}`,
    }));
    console.log(result.total);
    if (e.target.value === 'AC') {
      setState(() => ({
        display: '0',
      }));
    } else if (e.target.value === '=') {
      setState(() => ({
        display: result.total,
      }));
    }
    //  else {
    //   const { display } = state;
    //   setState(() => ({
    //     display: display === '0' ? `${`${e.target.value}`}` : `${display} ${e.target.value}`,
    //   }));
    // }
  };
  return (
    <div className="body">
      <button className="btn-input" type="button" value="0">{state.display}</button>
      <button type="button" onClick={handleClick} value="AC">AC</button>
      <button type="button" onClick={handleClick} value="+/-">+/-</button>
      <button type="button" onClick={handleClick} value="%">%</button>
      <button className="color" type="button" onClick={handleClick} value="รท">รท</button>
      <button type="button" onClick={handleClick} value="7">7</button>
      <button type="button" onClick={handleClick} value="8">8</button>
      <button type="button" onClick={handleClick} value="9">9</button>
      <button className="color" type="button" onClick={handleClick} value="x">ร</button>
      <button type="button" onClick={handleClick} value="4">4</button>
      <button type="button" onClick={handleClick} value="5">5</button>
      {' '}
      <button type="button" onClick={handleClick} value="6">6</button>
      <button className="color" type="button" onClick={handleClick} value="-">-</button>
      <button type="button" onClick={handleClick} value="1">1</button>
      <button type="button" onClick={handleClick} value="2">2</button>
      <button type="button" onClick={handleClick} value="3">3</button>
      <button className="color" type="button" onClick={handleClick} value="+">+</button>
      <button className="btn-0" type="button" onClick={handleClick} value="0">0</button>
      {' '}
      <button type="button" onClick={handleClick} value=".">.</button>
      <button className="color" type="button" onClick={handleClick} value="=">=</button>
    </div>
  );
}
