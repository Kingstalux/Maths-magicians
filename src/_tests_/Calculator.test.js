import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('matches quote snapshot', () => {
  const calculator = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(calculator).toMatchSnapshot();
});
