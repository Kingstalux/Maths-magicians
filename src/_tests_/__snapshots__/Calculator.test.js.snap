// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches quote snapshot 1`] = `
<div
  className="calculator-page"
>
  <h1>
    Lets do some maths
  </h1>
  <div
    className="body"
  >
    <button
      className="btn-input"
      type="button"
      value="0"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="color"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="color"
      onClick={[Function]}
      type="button"
      value="x"
    >
      ร
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
     
    <button
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="color"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="color"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="btn-0"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
     
    <button
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="color"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
